autogluon/utils/tabular/ml/utils.py

```diff
@@ -120,6 +120,34 @@ def dd_list():
     return defaultdict(list)
 
 
+def get_leaderboard_breakpoints(leaderboard: DataFrame, score_col='score_val', inference_time_col='pred_time_val_full') -> DataFrame:
+    """
+    Given a set of models, returns in ranked order from best score to worst score models which satisfy the criteria:
+    1. No other model in the set has both a lower inference time and a better or equal score.
+
+    :param leaderboard: Leaderboard DataFrame of model info containing score_col and inference_time_col
+    :param score_col: Column name in leaderboard of model score values
+    :param inference_time_col: Column name in leaderboard of model inference times
+    :return: Subset of the original leaderboard DataFrame containing only models that are a valid optimal choice at different valuations of score and inference time.
+    """
+    leaderboard = leaderboard.sort_values(by=[score_col, inference_time_col], ascending=[False, True]).reset_index(drop=True)
```

Comment:
should we just always use this sorting in `leaderboard()` (rather than only in `get_leaderboard_breakpoints`)?

Reply:
We do, however this is sorted because it has to be sorted for the function to work properly, and we shouldn't trust the input to be sorted.